### PR TITLE
[docs] Update ESIMD docs: SYCL-ESIMD co-existence

### PR DESCRIPTION
**Pull request by DenisBakhvalov**
This patch updates ESIMD documentation after implementation
of SYCL-ESIMD co-existence feature was completed.

### CODE REVIEW

**Comment by DenisBakhvalov** [comment]
@kbobrovs , please review.

**Comment by DenisBakhvalov** [comment]
> @kbobrovs , please review.

@kbobrovs, ping

File: sycl/doc/extensions/ExplicitSIMD/README.md [hunk 1]
```diff
@@ -89,5 +91,5 @@ program behavior if violated.
 
 ##### Other restrictions:
 - Only Intel GPU device is supported
-- Usual and ESIMD DPC++ kernels can not co-exist in the same application in most
-  cases
+- Interoperability between regular SYCL and ESIMD kernels is not supported.
+  I.e., it's not possible to call ESIMD code from SYCL and vise-versa.
```
Comment (kbobrovs) [inline sycl/doc/extensions/ExplicitSIMD/README.md:95]: We can in fact call SYCL functions with bodies from ESIMD, so please rephrase.

Comment (DenisBakhvalov) [reply sycl/doc/extensions/ExplicitSIMD/README.md]: Rephrased.

**Review by kbobrovs — COMMENTED** [review]
Please update the phrase at the bottom as well:
"Interoperability between regular SYCL and ESIMD kernels is not yet supported. I.e., it's not possible to invoke an ESIMD kernel from SYCL kernel and vice-versa"

**Comment by DenisBakhvalov** [comment]
> Please update the phrase at the bottom as well:
> "Interoperability between regular SYCL and ESIMD kernels is not yet supported. I.e., it's not possible to invoke an ESIMD kernel from SYCL kernel and vice-versa"

Thanks, done.

**Review by kbobrovs — APPROVED** [review]

**Pull request merged**